quinn-proto/src/connection/mod.rs

```diff
@@ -757,23 +768,28 @@ where
         })
     }
 
-    /// Returns `true` if a space has outgoing data to send
-    fn space_can_send(&self, space_id: SpaceId) -> bool {
+    /// Returns `if a space has outgoing data to send and what the expected type of data is
+    fn space_can_send(&self, space_id: SpaceId) -> CanSendResult {
         if self.spaces[space_id].crypto.is_some() && self.spaces[space_id].can_send() {
-            return true;
+            return CanSendResult::AckOnly;
```

Comment:
I don't think this is correct, or at least it's misleading. `Space::can_send` checks lots of non-ACK stuff like stream control frames.

Reply:
That's true. Any idea for a better name?

Reply:
Ping on this one, since it seems to be the main outstanding question.

The main thing that comes to my mind now is `CanSendResult::NonApplicationData` or `CanSendResult::ManagementData` - but even that is not 100% right since path challenge/response are are part of OneRtt/ZeroRtt

Reply:
Thinking a bit more about it, we can also just change the type to

```
struct CanSendReesult {
   acks: bool,
   other: bool,
}
```

then it's also clear whether ACK-only was indicated or not.

Reply:
Seems like a pretty decent type, though maybe a name like `FrameTypes` would make more sense for it?

Reply:
I updated with `SendableFrames`. I don't think it's 100% perfect either, since it actually is sent at least since checks are short-circuiting. If ACKs are indicated, something else can be send too (`can_send()` doesn't look into 1RTT spaces - we can make it do, but that requires more work just to keep the debug assert around). Maybe I should just remove the assert to land the bugfix?